node/signer.go

```diff
@@ -0,0 +1,106 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package node
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/ava-labs/avalanchego/config/node"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls/signer/localsigner"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls/signer/rpcsigner"
+	"github.com/ava-labs/avalanchego/utils/perms"
+)
+
+// NewStakingSigner returns a BLS signer based on the provided configuration.
+func NewStakingSigner(
+	config any,
+) (bls.Signer, error) {
+	switch cfg := config.(type) {
+	case node.EphemeralSignerConfig:
+		signer, err := localsigner.New()
+		if err != nil {
+			return nil, fmt.Errorf("couldn't generate ephemeral signer: %w", err)
```

Comment:
nit: in these error messages can we avoid special characters (aside from `:` which is common in go)? It makes it hard to parse logs with unix tools, since you have to escape special characters (so to be specific my feedback is to change "couldn't" to "could not").